### DISCUSSION TITLE
How can we customize the creation process of a relation manager?

### DISCUSSION THREAD

**Discussion by StaffCollab** [filamentphp/filament#7731] [Help]
On a resource, we have a `handleRecordCreation()` method we can use to override default creation behavior. How can we do something similar with a relations manager?
I tried the `using()` method without success.

Any suggestions?

**Comment by StaffCollab** [comment]
I solved it by using a custom action instead of the regular `CreateAction::make(),` but I still think there is an easier way of doing it.

**Comment by danharrin** [comment]
https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#customizing-the-creation-process

**Comment by WhiskeyM1ke** [comment]
I don't understand where you put `CreateAction::make()`

**Reply by leandrocfe** [reply]
Check this action in the XxRelationManager.php file.

**Reply by codespirit49** [reply]
I have the same question as you. Did you solve your problem @WhiskeyM1ke

**Reply by injektion** [reply]
If you want it to apply to the create/make button in the header action of a XxxRelationManager.php:

```
public function table(Table $table): Table
    {
            ->headerActions([
                Tables\Actions\CreateAction::make()->mutateFormDataUsing(function (array $data): array {
                    $data['active'] = 0;                
                    return $data;
                })
            ])

**Comment by abrardev99** [comment]
Still custom action is the way to go for me. 
```
Tables\Actions\Action::make('Associate')
                    ->label('Associate Attributes')
                    ->form([
                        Forms\Components\Select::make('attribute_id')
                            ->required()
                            ->searchable()
                            ->preload()
                            ->multiple()
                            ->options(fn() => Attribute::query()->pluck('name', 'id')->toArray())
                    ])
                    ->action(function (array $data) {
                        foreach ($data['attribute_id'] as $attributeId) {
                            $this->ownerRecord->attributes()->create([
                                'attribute_id' => $attributeId,
                                'product_form_tab_id' => $this->ownerRecord->product_form_tab_id
                            ]);
                        }
                    })
```

**Comment by MohannedF16** [comment]
Hello everyone , am new to filament and am having troubles with handling multiple videos uploads, my idea is when the admin uploads more than one video(say 5 videos) those videos should be inserted in the videos table (each video is a separate record) . can't accomplish that .
Any help would be very appreciated guys.
